### ISSUE TITLE
lb4 cli's generated modules on Windows have incomplete reference paths (minor)

### ISSUE THREAD

**Issue by rodolph66** [strongloop/loopback-next#5083]
**Steps to reproduce**
Simply follow the basic tutorial on the home page (or Todo tutorial in docs)

**Current Behavior**
Upon running npm start, the terminal reports errors related to modules not found

**Expected Behavior**
The app should compile and run

**Additional information**
This is a minor issue, I was able to go into each of the files and complete the "import" path for each of the generated modules then run successfully the app. It is just the inconvenience of it. Also other beginners might suffer.

Sample Output Image (of npm start):
![lb4_output](https://user-images.githubusercontent.com/9017114/78965484-de215200-7b05-11ea-9af2-10abef06a346.jpg)

Terminal output on running npm start on completion of the Todo/TodoList tutorial:

> todo-list@1.0.0 prestart F:\loopback\2020\official\todo-list
> npm run build

> todo-list@1.0.0 build F:\loopback\2020\official\todo-list
> lb-tsc

src/controllers/todo-list-image-todo-list.controller.ts:12:8 - error TS2307: Cannot find module '../models'.

12 } from '../models';
          ~~~~~~~~~~~

src/controllers/todo-list-image-todo-list.controller.ts:13:39 - error TS2307: Cannot find module '../repositories'.

13 import {TodoListImageRepository} from '../repositories';
                                         ~~~~~~~~~~~~~~~~~

src/controllers/todo-list-todo-list-image.controller.ts:21:8 - error TS2307: Cannot find module '../models'.

21 } from '../models';
          ~~~~~~~~~~~

src/controllers/todo-list-todo-list-image.controller.ts:22:34 - error TS2307: Cannot find module '../repositories'.

22 import {TodoListRepository} from '../repositories';
                                    ~~~~~~~~~~~~~~~~~

src/controllers/todo-list-todo.controller.ts:21:8 - error TS2307: Cannot find module '../models'.

21 } from '../models';
          ~~~~~~~~~~~

src/controllers/todo-list-todo.controller.ts:22:34 - error TS2307: Cannot find module '../repositories'.

22 import {TodoListRepository} from '../repositories';
                                    ~~~~~~~~~~~~~~~~~

src/controllers/todo-list.controller.ts:21:24 - error TS2307: Cannot find module '../models'.

21 import {TodoList} from '../models';
                          ~~~~~~~~~~~

src/controllers/todo-list.controller.ts:22:34 - error TS2307: Cannot find module '../repositories'.

22 import {TodoListRepository} from '../repositories';
                                    ~~~~~~~~~~~~~~~~~

src/controllers/todo-todo-list.controller.ts:12:8 - error TS2307: Cannot find module '../models'.

12 } from '../models';
          ~~~~~~~~~~~

src/controllers/todo-todo-list.controller.ts:13:30 - error TS2307: Cannot find module '../repositories'.

13 import {TodoRepository} from '../repositories';
                                ~~~~~~~~~~~~~~~~~

src/repositories/todo-list-image.repository.ts:2:63 - error TS2307: Cannot find module '../models'.

2 import {TodoListImage, TodoListImageRelations, TodoList} from '../models';
                                                                ~~~~~~~~~~~

src/repositories/todo-list-image.repository.ts:3:28 - error TS2307: Cannot find module '../datasources'.

3 import {DbDataSource} from '../datasources';
                             ~~~~~~~~~~~~~~~~

src/repositories/todo-list.repository.ts:2:64 - error TS2307: Cannot find module '../models'.

2 import {TodoList, TodoListRelations, Todo, TodoListImage} from '../models';
                                                                 ~~~~~~~~~~~

src/repositories/todo-list.repository.ts:3:28 - error TS2307: Cannot find module '../datasources'.

3 import {DbDataSource} from '../datasources';
                             ~~~~~~~~~~~~~~~~

src/repositories/todo.repository.ts:5:24 - error TS2307: Cannot find module '../models'.

5 import {TodoList} from '../models';
                         ~~~~~~~~~~~


Found 15 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! todo-list@1.0.0 build: `lb-tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the todo-list@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Bob\AppData\Roaming\npm-cache\_logs\2020-04-10T05_06_27_110Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! todo-list@1.0.0 prestart: `npm run build`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the todo-list@1.0.0 prestart script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Bob\AppData\Roaming\npm-cache\_logs\2020-04-10T05_06_27_174Z-debug.log

F:\loopback\2020\official\todo-list>

**Comment by raymondfeng** [comment]
It might have been fixed by https://github.com/strongloop/loopback-next/pull/5073. `yeoman-generator@4.8.0` introduces a breaking change. Can you check the following:

```
npm -g root
cd <dir-from-above>\@loopback\cli
npm ls yoeman-generator
```

If it shows 4.8.0, please revert it back to 4.7.2 as follows:
npm i yeoman-generator@4.7.2

Once #5073 is released, the problem should be fixed.

**Comment by raymondfeng** [comment]
Hmm, I can confirm that `npm i -g @loopback/cli` installs an incompatible version of `yeoman-generator`:

```
npm ls yeoman-generator
@loopback/cli@2.3.0 /Users/rfeng/.nvm/versions/node/v12.16.1/lib/node_modules/@loopback/cli
├─┬ yeoman-environment@2.9.0
│ └── yeoman-generator@4.8.0  deduped
└── yeoman-generator@4.8.0 
```

We'll publish a new release early next week.

**Comment by rodolph66** [comment]
Great....
All best....

Rodolph



Sent from myMail for iOS


Friday, 10 April 2020, 19:02 +0300 from notifications@github.com  <notifications@github.com>:

**Comment by raymondfeng** [comment]
It should be fixed by @loopback/cli@2.3.1 now. Please run `npm i -g @loopback/cli` to upgrade.
